homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -1 +1,120 @@
 """Support for Xiaomi Miio."""
+import asyncio
+import binascii
+import logging
+import socket
+from typing import List
+
+from construct import ConstructError
+from miio import Device, Message
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_NAME, CONF_TOKEN
+from homeassistant.helpers import discovery
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_MODEL = "model"
+CONFIG_SCHEMA = vol.Schema({}, extra=vol.ALLOW_EXTRA)
+
+DOMAIN = "xiaomi_miio"
+
+DISCOVERY_SOCKET_ADDR = ("<broadcast>", 54321)
+DISCOVERY_TIMEOUT = 3  # seconds
+DISCOVERY_HELLO_MESSAGE = bytes.fromhex(
+    "21310020ffffffffffffffffffffffffffffffffffffffffffffffffffffffff"
```

Comment:
Discovery should be added to the miio PyPI package.

Reply:
Thanks for your PR @vladimir-kotikov ! As balloob said, the discovery needs to be done using the methods already available in python-miio package. If there is something that is unclear, or if the API is not suitable we can adjust it, in fact, there is a WIP PR regarding to cleaning up the API: https://github.com/rytilahti/python-miio/pull/437 .